src/shogun/base/SGObject.h

```diff
@@ -123,6 +123,60 @@ class CSGObject
 
 	/** destructor */
 	virtual ~CSGObject();
+        
+        /** cereal serialization saver wrapper.
```

Comment:
This is *user documentation*
Please tune it more. Don't talk about wrappers or anything, just tell the user that this serializes the SGObject to a binary file. You can reference to other methods in doxygen for details